lib/query.js

```diff
@@ -2380,13 +2380,21 @@ Query.prototype.merge = function(source) {
   }
 
   opts.omit = {};
-  if (this._conditions && this._conditions.$and && source.$and) {
+  if (source.$and) {
     opts.omit['$and'] = true;
-    this._conditions.$and = this._conditions.$and.concat(source.$and);
+    if (this._conditions && this._conditions.$and) {
+      this._conditions.$and = this._conditions.$and.concat(source.$and);
+    } else {
+      this._conditions.$and = [...source.$and];
+    }
```

Comment:
This is a decent point. `this._conditions` should not be null, [we initialize it to empty object here](https://github.com/mongoosejs/mquery/blob/e7a8a3099ef282a2d9af36a2880981871adfe9fd/lib/mquery.js#L36-L38). However, users can do `query.setQuery(null)` which would break this logic. So I'll merge this.

Reply:
Ah, makes sense! Probably should make similar changes to the `$or` block below?

Reply:
Actually three things:
1. Should we make similar changes to the `$or` block below?
2. Do we need similar changes to [this](https://github.com/Automattic/mongoose/blob/d5202fbd1b7944c1d777ea18945efdee7716f42e/lib/query.js#L2341-L2348) section?
3. There's probably opportunity to put the `if (!this._conditions) this._conditions = {};` line at a higher level so we don't need it multiple times, since it impacts both `$and` and `$or` here and both of them within the `source instanceof Query` section.